src/database/index.tsx

```diff
@@ -123,6 +119,68 @@ const buildUpdatePayload = (entity: any) => {
   return payload;
 };
 
+const timestampToDate = (timestamp: firebase.firestore.Timestamp): Date => {
+  return timestamp.toDate();
+};
+
+const buildPlannedRelease = (plannedReleaseData: any): PlannedRelease => {
+  let plannedRelease: PlannedRelease = plannedReleaseData;
+
+  const plannedReleaseDataDate = plannedReleaseData.date;
+  if (plannedReleaseDataDate instanceof firebase.firestore.Timestamp) {
+    plannedRelease.date = timestampToDate(plannedReleaseDataDate);
+  } else {
+    plannedRelease.date = parseISO(plannedReleaseDataDate);
+  }
+
+  return plannedRelease;
+};
+
+const buildModelInputs = (document: any): ModelInputs => {
+  let modelInputs: ModelInputs = document;
+
+  modelInputs.observedAt = timestampToDate(document.observedAt);
+  modelInputs.updatedAt = timestampToDate(document.updatedAt);
+
+  const plannedReleases = document.plannedReleases;
+  if (plannedReleases) {
+    modelInputs.plannedReleases = plannedReleases.map(
+      (plannedReleaseData: any) => {
+        return buildPlannedRelease(plannedReleaseData);
+      },
+    );
+  }
+
+  return modelInputs;
+};
+
+const buildFacility = (
+  scenarioId: string,
+  document: firebase.firestore.DocumentData,
+): Facility => {
+  const documentData = document.data();
+
+  let facility: Facility = documentData;
+  facility.id = document.id;
+  facility.scenarioId = scenarioId;
+  facility.createdAt = timestampToDate(documentData.createdAt);
+  facility.updatedAt = timestampToDate(documentData.updatedAt);
+  facility.modelInputs = buildModelInputs(documentData.modelInputs);
```

Comment:
Do we need to handle `modelVersions`?

Reply:
We do need to handle `modelVersions`, but we don't need to handle them here.  Data returned from Firestore is "shallow" meaning that sub-collections (i.e. `modelVersions`) are not returned.  Note that we recently added a method for grabbing the model versions called `getFacilityModelVersions` and I did update that code to make use of this new `buildModelInputs` transformer.  The context is lost in this PR because it's just a one-line update, but you can see the whole method in all its glory here (see line 369):  https://github.com/Recidiviz/covid19-dashboard/blob/dec45a60e42f95c206efdb91f62564f6f40f22fd/src/database/index.tsx#L349-L378

Reply:
Right on! I did notice that you had updated `getFacilityModelVersions` but what I did _not_ know was that these returns were all shallow (or rather, I forgot, after the last time you told me). This makes sense. Thanks!